### PR TITLE
Enhance/press7 60 jetpack boost

### PR DESCRIPTION
**Pull request by AleTorrisi**
## Proposed changes

This branch introduces the new section "Advanced settings" which includes main Jetpack boost options, so that they can be set by Bluehost panel and synchronized with Jetpack panel too.

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Video

https://github.com/user-attachments/assets/f1251f9e-d4fc-403d-9c74-eb4fe1bdb384

## Checklist

- [x ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [x] I have viewed my change in a web-browser
- [x ] Linting and tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        url: apiUrl,
+        method: "POST",
+        headers: { "X-NFD-INSTALLER": INSTALL_TOKEN },
+        data: { plugin, activate: true, queue: false },
+      });
+      setIsActive(true);
+      setModuleStatus(true);
+      setMessage( constants.text.jetpackBoostInstalling );
+    } catch (error) {
+      console.log(error.message);
```
Comment (wpscholar) [inline components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js:29]: Should this be a `console.error()` instead?

File: components/cacheSettings/index.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Container, RadioGroup } from "@newfold/ui-component-library";
+import { Container, RadioGroup, Alert } from "@newfold/ui-component-library";
```
Comment (wpscholar) [inline components/cacheSettings/index.js:1]: I don't see `Alert` used in this file.

File: components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js (hunk 1 above)
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js]: That's right, I replaced the function and committed the change

File: components/cacheSettings/index.js (hunk 2 above)
Comment (AleTorrisi) [reply components/cacheSettings/index.js]: I removed the unused component, it's committed

**Comment by chrisdavidmiles** [comment]
@arnavchaudhary We need a PR on this

File: components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  const [isLoading, setIsLoading] = useState(false);
+  const [isActive, setIsActive] = useState(false);
+  const [message, setMessage] = useState(null);
+
+  const handleInstallActivate = async () => {
+    setIsLoading(true);
+    setMessage( constants.text.jetpackBoostInstalling );
+
+    const apiUrl = methods.NewfoldRuntime.createApiUrl("/newfold-installer/v1/plugins/install");
+    const INSTALL_TOKEN = methods.NewfoldRuntime.ecommerce.install_token;
```
Comment (arunshenoy99) [inline components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js:15]: We should avoid using the `ecommerce` method dependencies within the performance module. Instead, let's align it with the runtime filter approach as per one of my previous comments for better consistency and modularity.

**Review by arunshenoy99 — CHANGES_REQUESTED** [review]
Hey @AleTorrisi, thank you for the PR! This looks fantastic, especially since it's your first time working on this codebase. I added a few small comments and optimizations. Please take a look and let me know if they make sense.

Other than that:
1. Let's ensure we run the `wp-scripts` linter on the JS files.
2. Let's ensure we have a newline at the end of each file for consistency.
3. Here's what I am seeing on my local environment with your changes. If I’m missing something, that’s all OK, but please reconfirm if these buttons look correct:
   
   image

Let me know your thoughts or if anything needs further clarification ✌️

File: components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { Spinner } from "@newfold/ui-component-library";
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js]: Commit [94f7456cc33f695a4b808c494d6fac79d3eb66e6](https://github.com/newfold-labs/wp-module-performance/commit/94f7456cc33f695a4b808c494d6fac79d3eb66e6)

File: components/performance/index.js [hunk 5]
```diff
@@ -45,12 +49,18 @@ const Performance = ({methods, constants, Components, ...props}) => {
...
                 />
             </Container.Block>
-            <Container.Block className={'newfold-clear-cache'}>
+            <Container.Block separator={true} className={'newfold-clear-cache'}>
                 <ClearCache
                     methods={methods}
                     constants={constants}
                 />
             </Container.Block>
+            <Container.Block separator={true} className={'newfold-cache-advanced-settings'}>
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/performance/index.js]: Commit [d4c407407e5586d462a0e193c5524c14ce318c00](https://github.com/newfold-labs/wp-module-performance/commit/d4c407407e5586d462a0e193c5524c14ce318c00)

File: includes/Performance.php [hunk 6]
```diff
@@ -57,6 +57,15 @@ class Performance {
...
 	protected $container;
 
+	/**
+	 * Array map of API controllers.
+	 *
+	 * @var array
+	 */
+	protected $controllers = array(
+		'NewfoldLabs\\WP\\Module\\Performance\\RestApi\\JetpackController',
+	);
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply includes/Performance.php]: Commit: [319e8c9f8c20b9a5433eeec021df33b16714425d](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/319e8c9f8c20b9a5433eeec021df33b16714425d)

File: includes/Performance.php [hunk 7]
```diff
@@ -80,6 +89,10 @@ public function __construct( Container $container ) {
 		$container->set( 'cachePurger', $cachePurger );
 
 		$container->set( 'hasMustUsePlugin', file_exists( WPMU_PLUGIN_DIR . '/endurance-page-cache.php' ) );
+
+		add_action( 'admin_enqueue_scripts', array( $this, 'register_scripts' ) );
+
+		add_action( 'rest_api_init', array( $this, 'register_routes' ) );
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply includes/Performance.php]: Commit: [319e8c9f8c20b9a5433eeec021df33b16714425d](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/319e8c9f8c20b9a5433eeec021df33b16714425d)

File: includes/RestApi/JetpackController.php [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+		$this->container = $container;
+	}
+
+	/**
+	 * Register API routes.
+	 *
+	 * @return void
+	 */
+	public function register_routes() {
+		\register_rest_route(
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply includes/RestApi/JetpackController.php]: Commit: [319e8c9f8c20b9a5433eeec021df33b16714425d](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/319e8c9f8c20b9a5433eeec021df33b16714425d)

File: includes/RestApi/JetpackController.php [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+			$this->namespace,
+			$this->rest_base . '/get_options',
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply includes/RestApi/JetpackController.php]: Commit: [319e8c9f8c20b9a5433eeec021df33b16714425d](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/319e8c9f8c20b9a5433eeec021df33b16714425d)

File: includes/RestApi/JetpackController.php [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+			array(
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_options' ),
+				'permission_callback' => function () {
+					return 'manage_options';
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply includes/RestApi/JetpackController.php]: Commit: [319e8c9f8c20b9a5433eeec021df33b16714425d](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/319e8c9f8c20b9a5433eeec021df33b16714425d)

File: includes/RestApi/JetpackController.php [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+				$data  = update_option( 'jetpack_boost_ds_' . $id, $value );
+			} else {
+				$data = update_option( 'jetpack_boost_status_' . $field['id'], $field['value'] );
+			}
+		}
+		return new \WP_REST_Response(
+			array( $data ),
+			200
+		);
+	}
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply includes/RestApi/JetpackController.php]: Commit: [23ed28b7611838482cf64d8bcf59ccd7f2fb1318](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/23ed28b7611838482cf64d8bcf59ccd7f2fb1318)

File: components/advancedSettings/JetpackBoost/index.js [hunk 12]
```diff
@@ -0,0 +1,156 @@
...
+          label: constants.text.jetpackBoostExcludeCssTitle,
+          description: '',
+          value: '',
+          type: 'textarea',
+        }
+      ]
+    }
+  ]);
+
+  const style = {};
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js]: Commit [6bf3faf9f55c63a4479634ea6d2cd8f8dd53e56b](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/6bf3faf9f55c63a4479634ea6d2cd8f8dd53e56b)

File: components/advancedSettings/JetpackBoost/index.js [hunk 13]
```diff
@@ -0,0 +1,156 @@
...
+
+  const [is_module_active, setModuleStatus] = useState(false);
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js]: Commit [a569d2a5cc96f6d297d8a024bcb1ec3564cad297](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/a569d2a5cc96f6d297d8a024bcb1ec3564cad297)

File: components/advancedSettings/JetpackBoost/index.js [hunk 14]
```diff
@@ -0,0 +1,156 @@
...
+
+
+  const [loading, setLoading] = useState(true); // Nuovo stato per il caricamento
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js:68]: Commit [a569d2a5cc96f6d297d8a024bcb1ec3564cad297](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/a569d2a5cc96f6d297d8a024bcb1ec3564cad297)

File: components/advancedSettings/JetpackBoost/index.js [hunk 15]
```diff
@@ -0,0 +1,156 @@
+import React, { useEffect, useState } from 'react';
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js]: Commit [3e39933547731af0d6ca79f44dda8c387a2e53d8](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/a569d2a5cc96f6d297d8a024bcb1ec3564cad297)

File: components/advancedSettings/JetpackBoost/index.js [hunk 16]
```diff
@@ -0,0 +1,156 @@
...
+
+        setFields(newFields)
+        setModuleStatus(response.is_module_active)
+        setLoading(false)
+      })
+      .catch((error) => {
+
+        setLoading(false)
+      })
+  }, [is_module_active])
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js]: Commit [be0bf0233b0da9deca7aa75809b0445dc980cb30](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/be0bf0233b0da9deca7aa75809b0445dc980cb30)

File: components/advancedSettings/JetpackBoost/index.js [hunk 17]
```diff
@@ -0,0 +1,156 @@
...
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <>
+      {!is_module_active ? (
+        <div className="nfd-container-upsell" >
+          <InstallActivatePluginButton methods={methods} constants={constants} setModuleStatus={setModuleStatus} />
+          <FeatureUpsell
+            cardText={__('Enjoy with Jetpack Boost module', 'wp-module-performance')}
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js]: Commit [d1c6e62f0f210e3aaa3028c02b84f76a69672e22](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/d1c6e62f0f210e3aaa3028c02b84f76a69672e22)

File: components/advancedSettings/JetpackBoost/index.js [hunk 18]
```diff
@@ -0,0 +1,156 @@
...
+            cardLink='https://wordpress.org/plugins/jetpack-boost/'
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js]: Commit: [b71cd5f83b37bec4e790f3e4cfc5dc0adb2e98e7](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/b71cd5f83b37bec4e790f3e4cfc5dc0adb2e98e7)

File: components/advancedSettings/JetpackBoost/index.js [hunk 19]
```diff
@@ -0,0 +1,156 @@
...
+                      <SingleOption params={subfield} isChild={true} methods={methods} constants={constants} />
+                    </div>
+                  ))}
+                </>
+              )}
+            </div>
+          ))}
+        </>
+      )}
+    </>
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js]: Commit: [b71cd5f83b37bec4e790f3e4cfc5dc0adb2e98e7](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/b71cd5f83b37bec4e790f3e4cfc5dc0adb2e98e7)

File: components/advancedSettings/JetpackBoost/index.js [hunk 20]
```diff
@@ -0,0 +1,156 @@
...
+              return (
+                <SingleOption key={field.id} params={field} methods={methods} constants={constants} />
+              );
+            })}
+          </FeatureUpsell>
+        </div>
+      ) : (
+        <>
+          {fields.map((field) => (
+            <div className="nfd-container-single-option" style={{ marginBottom: "20px" }} key={field.id}>
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/index.js]: Commit [b58793df9151066cfa2ac50669b1bf05cff8a753](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/b58793df9151066cfa2ac50669b1bf05cff8a753)

File: components/performance/index.js [hunk 21]
```diff
@@ -1,6 +1,10 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply components/performance/index.js]: Commit [b58793df9151066cfa2ac50669b1bf05cff8a753](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/b58793df9151066cfa2ac50669b1bf05cff8a753)

File: includes/Performance.php [hunk 22]
```diff
@@ -142,6 +155,31 @@ function () {
...
 	}
 
+
+	/**
+	 * Register admin scripts for module.
+	 *
+	 * @return void
+	 */
+	public function register_scripts() {
+		\wp_enqueue_style( 'nfd-performance', plugin_dir_url( __DIR__ ) . '/src/css/style.css', null, '1', 'screen' );
```
[the remark this answers is not in the document]
Comment (AleTorrisi) [reply includes/Performance.php]: Commit [b58793df9151066cfa2ac50669b1bf05cff8a753](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/b58793df9151066cfa2ac50669b1bf05cff8a753)

File: components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js (hunk 3 above)
Comment (AleTorrisi) [reply components/advancedSettings/JetpackBoost/InstallActivatePluginButton.js]: Commit [717557b46d8ed29411c29cd233773a943764c768](https://github.com/newfold-labs/wp-module-performance/pull/25/commits/717557b46d8ed29411c29cd233773a943764c768)

**Review by arunshenoy99 — DISMISSED** [review]
I think the performance test is failing for something related to accessibility. Other than that looks good to me. Thanks! @AleTorrisi

File: components/performance/defaultText.js [hunk 23]
```diff
@@ -1,27 +1,135 @@
...
+	performanceAdvancedSettingsTitle: __(
+		'Advanced settings',
+		'wp-module-performance'
+	),
+	performanceAdvancedSettingsDescription: __(
+		'Additional speed and scalability features powered by Jetpack Boost to make your site as fast as it can be.',
+		'wp-module-performance'
+	),
+	jetpackBoostCriticalCssTitle: __(
+		'Optimize Critical CSS Loading (manual)',
```
Comment (bradp) [inline components/performance/defaultText.js:76]: What's "manual" mean in this case? cc @chrisdavidmiles

Comment (bradp) [reply components/performance/defaultText.js:76]: Might be the difference between paid Jetpack Boost vs free - paid this is automatic.

**Review by arunshenoy99 — DISMISSED** [review]

**Review by arunshenoy99 — APPROVED** [review]
Looks good, the test is failing because the workflow is not picking up the new installer dependency. This should not be an issue when we create a bump PR in the web/mojo plugin.

**Pull request merged**